### ISSUE TITLE
Server API: POST and GET transactions

### ISSUE THREAD

**Issue by diminator** [bigchaindb/bigchaindb#554]
See #362 

### Post a transaction 
```
POST /transactions/

PAYLOAD
    <TX>

RETURNS
    201 + <ID>
    {4xx, 5xx} + <reason>
``` 
BigchainDB foresees `bigchaindb.write_transaction(tx)` to make the call to the DB

#### Questions:
- Does this post do a multicast or simply `bigchaindb.write_transaction(tx)`?
- see #347: `CREATE` tx's require a fulfillment from the federation: ie.  `bigchaindb.sign_transaction(tx, private_key=bigchain.me_private)` is called in the `POST /transactions/` endpoint. In the future we want to split this up: 
  1. retrieve the permission first (ie a separate call to get the fulfillment from the federation)
  2. POST the signed tx.

### Retrieve a transaction
```
GET /transactions/<ID>

RETURNS
    200 + <TX> 
    404
```
BigchainDB foresees `bigchaindb.get_transaction(id)` to make the call to the DB

#### Questions: 

- does the call only looks to the `bigchain` blocks or also in `backlog`?
- does it also look into `bigchain` blocks that are `UNDECIDED`?

**Comment by TimDaub** [comment]
> Does this post do a multicast or simply bigchaindb.write_transaction(tx)?

I think for now we don't need to do a multicast.

**Comment by sohkai** [comment]
> Does this post do a multicast or simply bigchaindb.write_transaction(tx)?

Sylvain and I had a brief discussion about this before and I'm sure you guys have had similar ones. Supporting multi-cast and multi-node connections should be their own projects; their scopes are way too big.

> #### Questions:
    does the call only looks to the bigchain blocks or also in backlog?
    does it also look into bigchain blocks that are UNDECIDED?

From the discussion with the `/status` endpoint, I think we should only return transactions that are DECIDED and in the chain. Otherwise, a 404 (and maybe a url link to check the transaction's status, depending how hyperlinked we want to get).

**Comment by r-marques** [comment]
> Does this post do a multicast or simply bigchaindb.write_transaction(tx)?

This will probably be done in the future but it needs to be done on the client side. The client should contact multiple servers so that a rogue server does not simply drop the transactions. We need a protocol for this.

Agree with the second point. We can even have an explicit threshold fulfillment in a `CREATE` transaction so it requires a signature from both the `federation node` and the `new_owner` so that you cannot create assets to random pub keys.

The flow would be:
1. Send a `CREATE` request with some payload
2. Receive the `CREATE` transaction fulfilled by the federation node
3. Add user fulfillment and send it.

**Comment by diminator** [comment]
> Questions:
> 
> does the call only looks to the bigchain blocks or also in backlog?
> does it also look into bigchain blocks that are UNDECIDED?

It seems that `bigchain.get_transaction(txid)` only looks for transactions in a `VALID` block on `bigchain`.
cc @rhsimplex can you verify?

**Comment by diminator** [comment]
> It seems that bigchain.get_transaction(txid) only looks for transactions in a VALID block on bigchain.

EDIT: the `UNDECIDED` ones are also in ther

**Comment by r-marques** [comment]
relates to #557

**Comment by diminator** [comment]
solved by #563
